WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -637,20 +692,8 @@ private extension ProductFormViewController {
 
 // MARK: - Navigation actions handling
 //
-extension ProductFormViewController {
-    override func shouldPopOnBackButton() -> Bool {
-        if viewModel.hasUnsavedChanges() {
-            presentBackNavigationActionSheet()
-            return false
-        }
-        return true
-    }
-
-    override func shouldPopOnSwipeBack() -> Bool {
-        return shouldPopOnBackButton()
-    }
-
-    private func presentBackNavigationActionSheet() {
+private extension ProductFormViewController {
+    func presentBackNavigationActionSheet() {
```

Comment:
Do you think it's worth moving this method next to the others (`shouldPopOnBackButton` and `shouldPopOnSwipeBack`)?
Right now we have navigation actions handling in two separate places.

Reply:
the functions are usually together but this is a special case, since @objc functions like `shouldPopOnBackButton` and `shouldPopOnSwipeBack` can't be in an extension of a class with generics. I think I'd prefer this staying in an extension rather than in the already crowded main class. lemme know if you have other concerns!